### PR TITLE
Added test utilities methods

### PR DESCRIPTION
**Pull request by slinkydeveloper**
* Added WaitForPodState to wait a state of a single pod
* Added WaitForServiceEndpoints

Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

### CODE REVIEW

File: test/kube_checks.go [hunk 1]
```diff
@@ -75,6 +75,55 @@ func WaitForPodListState(client *KubeClient, inState func(p *corev1.PodList) (bo
 	})
 }
 
+// WaitForPodState polls the status of the specified Pod
+// from client every interval until inState returns `true` indicating it
+// is done, returns an error or timeout. desc will be used to name the metric
+// that is emitted to track how long it took to get into the state checked by inState.
+func WaitForPodState(client *KubeClient, inState func(p *corev1.Pod) (bool, error), name string, namespace string) error {
+	p := client.Kube.CoreV1().Pods(namespace)
+	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForPodState/%s", name))
```
Comment (vagababov) [inline test/kube_checks.go]:
```suggestion
	span := logging.GetEmitableSpan(context.Background(), "WaitForPodState/"+name)
```

File: test/kube_checks.go [hunk 2]
```diff
@@ -75,6 +75,55 @@ func WaitForPodListState(client *KubeClient, inState func(p *corev1.PodList) (bo
...
+	defer span.End()
+
+	return wait.PollImmediate(interval, podTimeout, func() (bool, error) {
+		p, err := p.Get(name, metav1.GetOptions{})
+		if err != nil {
+			return true, err
```
Comment (vagababov) [inline test/kube_checks.go]:
```suggestion
			return false, err
```

File: test/kube_checks.go [hunk 3]
```diff
@@ -75,6 +75,55 @@ func WaitForPodListState(client *KubeClient, inState func(p *corev1.PodList) (bo
...
+		}
+		return inState(p)
+	})
+}
+
+// WaitForServiceHasAtLeastOneEndpoint polls the status of the specified Service
+// from client every interval until number of service endpoints = numOfEndpoints
+func WaitForServiceEndpoints(client *KubeClient, svcName string, svcNamespace string, numOfEndpoints int) error {
+	endpointsService := client.Kube.CoreV1().Endpoints(svcNamespace)
+	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForServiceHasAtLeastOneEndpoint/%s", svcName))
```
Comment (vagababov) [inline test/kube_checks.go]:
```suggestion
	span := logging.GetEmitableSpan(context.Background(), "WaitForServiceHasAtLeastOneEndpoint/"+ svcName)
```

File: test/kube_checks.go [hunk 4]
```diff
@@ -75,6 +75,55 @@ func WaitForPodListState(client *KubeClient, inState func(p *corev1.PodList) (bo
...
+	defer span.End()
+
+	return wait.PollImmediate(interval, podTimeout, func() (bool, error) {
+		endpoints, err := endpointsService.List(metav1.ListOptions{})
+		if err != nil {
+			return true, err
```
Comment (vagababov) [inline test/kube_checks.go]:
```suggestion
			return false, err
```

File: test/kube_checks.go [hunk 5]
```diff
@@ -75,6 +75,55 @@ func WaitForPodListState(client *KubeClient, inState func(p *corev1.PodList) (bo
...
+
+// WaitForServiceHasAtLeastOneEndpoint polls the status of the specified Service
+// from client every interval until number of service endpoints = numOfEndpoints
+func WaitForServiceEndpoints(client *KubeClient, svcName string, svcNamespace string, numOfEndpoints int) error {
+	endpointsService := client.Kube.CoreV1().Endpoints(svcNamespace)
+	span := logging.GetEmitableSpan(context.Background(), fmt.Sprintf("WaitForServiceHasAtLeastOneEndpoint/%s", svcName))
+	defer span.End()
+
+	return wait.PollImmediate(interval, podTimeout, func() (bool, error) {
+		endpoints, err := endpointsService.List(metav1.ListOptions{})
```
Comment (vagababov) [inline test/kube_checks.go]: just `endpointsService.Get` should suffice here and loop below should not be required.

File: test/kube_checks.go [hunk 6]
```diff
@@ -75,6 +75,55 @@ func WaitForPodListState(client *KubeClient, inState func(p *corev1.PodList) (bo
...
+			if e.Name == svcName && countEndpointsNum(&e) == numOfEndpoints {
+				return true, nil
+			}
+		}
+
+		return false, nil
+	})
+}
+
+func countEndpointsNum(e *corev1.Endpoints) int {
```
Comment (vagababov) [inline test/kube_checks.go]: I am pretty sure we have such a method somewhere, but it might be in serving only.

Comment (slinkydeveloper) [reply test/kube_checks.go]: I didn't found such function in pkg, I tried to seach a bit without results

File: test/kube_checks.go [hunk 7]
```diff
@@ -97,26 +97,27 @@ func WaitForPodState(client *KubeClient, inState func(p *corev1.Pod) (bool, erro
...
 		if err != nil {
-			return true, err
+			return false, err
 		}
 
-		for _, e := range endpoints.Items {
-			if e.Name == svcName && countEndpointsNum(&e) == numOfEndpoints {
-				return true, nil
-			}
+		if countEndpointsNum(endpoint) == numOfEndpoints {
```
Comment (vagababov) [inline test/kube_checks.go]: and the final change, please
`return countEndpointsNum(endpoint) == numOfEndpoints, nil`

**Pull request merged**